### PR TITLE
Update Dragen transcriptome tool and pipeline

### PR DESCRIPTION
**Pull request by skanwal**
Updated Dragen4-WTS pipeline to allow for bam inputs. This is needed for APGI data reprocessing on ICA. 

Successful test run in dev `wfr.7c11079c72b146bdb2925621e59d90f3`.
Currently testing the complete pipeline with `wfr.63744a0e5e1b4015baf2c4460d664885`

### CODE REVIEW

File: workflows/dragen-transcriptome-pipeline/4.0.3/dragen-transcriptome-pipeline__4.0.3.cwl [hunk 1]
```diff
@@ -47,6 +49,15 @@ inputs:
...
         * Read2File (optional)
     type: ../../../schemas/fastq-list-row/1.0.0/fastq-list-row__1.0.0.yaml#fastq-list-row[]?
+  # Option 3
+  bam_input:
+    label: bam input
+    doc: |
+      Input a BAM file for WTS analysis
+    type: File?
+    secondaryFiles:
+      - pattern: ".bam"
```
Comment (alexiswl) [inline workflows/dragen-transcriptome-pipeline/4.0.3/dragen-transcriptome-pipeline__4.0.3.cwl:59]: Pattern should be .bai no?

File: workflows/dragen-transcriptome-pipeline/4.0.3/dragen-transcriptome-pipeline__4.0.3.cwl [hunk 2]
```diff
@@ -80,6 +97,11 @@ inputs:
     doc: |
       Mark identical alignments as duplicates
     type: boolean
+  enable_sort:
+    label: enable sort
+    doc: |
+      True by default, only set this to false if using --bam-input and --tumor-bam-input parameters
```
Comment (alexiswl) [inline workflows/dragen-transcriptome-pipeline/4.0.3/dragen-transcriptome-pipeline__4.0.3.cwl:103]: Is --tumor-bam-input an option in Dragen WTS?

File: tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl [hunk 3]
```diff
@@ -103,12 +99,33 @@ inputs:
...
     type: ../../../schemas/fastq-list-row/1.0.0/fastq-list-row__1.0.0.yaml#fastq-list-row[]?
+    inputBinding:
+      prefix: "--fastq-list="
+      separate: False
+      valueFrom: "$(get_fastq_list_csv_path())"
+  # Option 3
+  bam_input:
+    label: bam input
+    doc: |
+      Input a normal BAM file for the variant calling stage
```
Comment (alexiswl) [inline tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl:121]: normal BAM file?

File: tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl [hunk 4]
```diff
@@ -136,6 +153,16 @@ inputs:
       prefix: "--output-directory="
       separate: False
   # Alignment options
+  enable_map_align:
+    label: enable map align
+    doc: |
+      Enabled by default.
+      Set this value to false if using bam_input AND tumor_bam_input
```
Comment (alexiswl) [inline tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl:160]: Is tumor_bam_input a thing in Dragen WTS?

**Review by alexiswl — CHANGES_REQUESTED** [review]

File: workflows/dragen-transcriptome-pipeline/4.0.3/dragen-transcriptome-pipeline__4.0.3.cwl (hunk 2 above)
Comment (skanwal) [reply workflows/dragen-transcriptome-pipeline/4.0.3/dragen-transcriptome-pipeline__4.0.3.cwl]: No. Carried over from somatic pipeline. Updated in https://github.com/umccr/cwl-ica/pull/347/commits/723a29ef3e0254e6288592eb31dd0d0e639f0209

File: tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl (hunk 3 above)
Comment (skanwal) [reply tools/dragen-transcriptome/4.0.3/dragen-transcriptome__4.0.3.cwl]: Updated in https://github.com/umccr/cwl-ica/pull/347/commits/723a29ef3e0254e6288592eb31dd0d0e639f0209

**Comment by skanwal** [comment]
Dragen transcriptome (v4) pipeline test run has completed `wfr.9e3f19f57d754d349637e63fca2c1e70`.

**Review by alexiswl — APPROVED** [review]

**Pull request merged**